### PR TITLE
BUG: RangeIndex can be created without args

### LINKED ISSUE #13793: ERR: disallow RangeIndex()
xref #13749 

this is not consistent with other Indexes where we don't allow them to be constructed with no data passed (it can be ``None``), but that's the idea. ``start`` must be not-None.

IOW this should raise ``ValueError``
```
In [3]: pd.RangeIndex()
Out[3]: RangeIndex(start=0, stop=0, step=1)
```

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #13793
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/types/missing.py [hunk 1]
```diff
@@ -143,8 +143,19 @@ def _isnull_ndarraylike(obj):
     elif is_datetimelike(obj):
         # this is the NaT pattern
         result = values.view('i8') == iNaT
+    elif is_object_dtype(dtype):
+        shape = values.shape
+        result = np.empty(shape, dtype=bool)
+        vec = lib.isnullobj(values.ravel())
+        result[...] = vec.reshape(shape)
```
Comment (jreback) [inline pandas/types/missing.py]: is this for this PR?

Comment (sinhrks) [inline pandas/types/missing.py]: Sorry, unrelated. Removed.

**Comment by jreback** [comment]
lgtm.

**Comment by jreback** [comment]
thanks @sinhrks

**Pull request closed without merging**